src/Controller/Backend/Invitation.php

```diff
@@ -0,0 +1,305 @@
+            }
+        }
+
+        //Generate token for invitation code
+        $token = bin2hex(openssl_random_pseudo_bytes(16));
+
+        $tokenEntity = new Entity\Tokens();
+
+        $tokenEntity->setToken($token);
+        $tokenEntity->setRoles($roles);
+        $tokenEntity->setExpiration($expiration);
+
+        $this->getRepository('Bolt\Storage\Entity\Tokens')->save($tokenEntity);
+
+        return $token;
+    }
+
+    /**
+     * Share the invitation code by email.
+     *
+     * @param Request $request The Symfony Request
+     *
+     * @return \Bolt\Response\BoltResponse|\Symfony\Component\HttpFoundation\RedirectResponse
+     */
+    public function sendLink(Request $request)
+    {
+        // Get all the necessary data to send the email
+        $to = $request->request->get('to');
+        $subject = $request->request->get('subject');
+        $text = $request->request->get('message');
+        $link = $request->request->get('link');
+
+        // Validate the form to send the invitation code by email
+        $book = array(
+            'to' => $to,
+            'subject' => $subject,
+            'text' => $text,
+        );
+
+        $constraint = new Assert\Collection(array(
+            'to' => new Assert\Email(),
+            'subject' => new Assert\NotBlank(),
+            'text' => new Assert\NotBlank(),
+        ));
+
+        $violationList = $this->app['validator']->validate($book, $constraint);
+
+        if (count($violationList) > 0) {
+            foreach ($violationList as $violation) {
+                $field = preg_replace('/\[|\]/', "", $violation->getPropertyPath());
+                $error = $violation->getMessage();
+                $errors[$field] = $error;
+            }
+            return $this->app->json($errors, 400);
+        }
+
+        // Get the current user to use his email on the "From" field.
+        $userEntity = new Entity\Users($this->getUser());
+
+        $from = $this->app['config']->get('general/mailoptions/senderMail', $userEntity->getEmail());
+
+        // Compile the email with the invitation link.
+        $mailhtml = $this->app['render']->render(
+            '@bolt/mail/invitation.twig',
+            [
+                'message' => $text,
+                'link' => $link,
+            ]
+        );
+
+        $message = $this->app['mailer']
+            ->createMessage('message')
+            ->setSubject($subject)
+            ->setFrom($from)
+            ->setReplyTo($from)
+            ->setTo($to)
+            ->setBody(strip_tags($mailhtml))
+            ->addPart($mailhtml, 'text/html');
+
+        $failed = true;
+        $failedRecipients = [];
+
+        try {
+            $recipients = $this->app['mailer']->send($message, $failedRecipients);
+
+            // Try and send immediately
+            $this->app['swiftmailer.spooltransport']->getSpool()->flushQueue($this->app['swiftmailer.transport']);
```

Comment:
This doesn't behave the way you might think. It still uses the spooler, and things like unavailable SMTP servers likely won't show up depending on PHP setting for script timeout, etc.

I've found the best way is to use 
```
$mailer = $app['mailer']->newInstance($app['swiftmailer.transport']);
$mailer->send( … );
```